css/selectors/focus-visible-007.html

```diff
@@ -10,98 +10,47 @@
   <script src="/resources/testdriver.js"></script>
   <script src="/resources/testdriver-vendor.js"></script>
   <style>
-    [data-hadkeydown] :focus-visible {
-      outline: darkgreen solid 5px;
+    @supports not (selector(:focus-visible)) {
+      :focus {
+        outline: red solid 5px;
+      }
     }
 
-    [data-hadmousedown] :focus-visible {
-      outline: red solid 5px;
+    :focus-visible {
+      outline: green solid 5px;
     }
 
-    [data-hadkeydown] :focus:not(:focus-visible) {
-      outline: 0;
-      background-color: tomato;
+    :focus:not(:focus-visible) {
+      outline: blue solid 5px;
     }
-
-    [data-hadmousedown] :focus:not(:focus-visible) {
-      outline: 0;
-      background-color: darkseagreen;
-    }
-
   </style>
 </head>
 <body>
   This test checks that using the keyboard in a way that does not move focus still causes <code>:focus-visible</code> matching to trigger.
   <ol id="instructions">
     <li>If the user-agent does not claim to support the <code>:focus-visible</code> pseudo-class then SKIP this test.</li>
-    <li>Use the mouse to focus the element below that says "Click me first."</li>
-    <li>If the element has a red outline, then the test result is FAILURE.</li>
```

Comment:
Why change this?

Reply:
I believe the test was kind of complex and testing 2 different things, that's why I split it in 2 different tests.
Then I don't need so many steps and elements on this first test, and that's why I removed this part.

Reply:
You changed this from a failure to a success, but the test doesn't pass unless both steps (initial focus from click not matching :focus-visible, and later pressing a key triggering :focus-visible matching) pass. So it's misleading to claim that the test is a success after step one. Why not leave this specific text the way it was?

Reply:
Sorry my previous comment was too quick and wrong.

The steps on the new version of the test are:
  <ol id="instructions">
    <li>If the user-agent does not claim to support the <code>:focus-visible</code> pseudo-class then SKIP this test.</li>
    <li>Use the mouse to focus the element below that says "Click me."</li>
    <li>If the element has a blue outline, then the test result is a SUCCESS.</li>
    <li>Press the SHIFT key.</li>
    <li>If the element now has green outline, then the test result is SUCCESS.</li>
  </ol>

The first time you click only `:focus` matches and not `:focus-visible`, and it has a blue outline.
Then after clicking SHIFT, `:focus-visible` matches and the outline changes to green.

I can come back to the previous colors I guess, but changing the test again. The thing was that I was having a hard time understanding the whole test and I believed the new version was better to understand (apart from the issues with the wrong assertions).

Reply:
I agree with splitting the test into two parts, apologies for not being clear about that.

Splitting this out into two tests means that the later part of the manual steps (click on the second element) are no longer needed, I agree with that.

However, previously the manual test made it clear that not passing the first part of the test (clicking the element shouldn't trigger :focus-visible matching) is an instant fail, meaning the second part of the test (pressing shift) is irrelevant. That's why it had the mechanism for setting up that red border to demonstrate that.

Can we have separate changes for splitting this out into two tests, and simplifying focus-visible-007.html? That might make this review easier for both of us.

Reply:
Ok, maybe I went too far away changing this test, I know understand your point. Sorry for all the back and forth.

I can separate this in different PRs, or maybe simplify this PRs by just removing the last steps in `focus-visible-007.html` but keep the rest (fixing the wrong assertions), and then add the proper manual instructions to `focus-visible-013.html`. Sorry for the mess.

Reply:
Thank you for your patience with all my comments! Even just splitting this change into two separate commits for the two different changes would help, or otherwise just keep this change as the splitting out (uncontroversial and we can land it right away), and we can work on the simplification/leak fixing etc. separately.

Reply:
So I've tried to just split this in two tests, but `focus-visible-007.html` has other problems too.

The test starts by sending a SHIFT key, then it sends the SHIFT key again after focus, and at that point it starts to test things. I believe I can fix `focus-visible-007.html` in a separated PR and keep this PR for adding a new test that covers part of `focus-visible-007.html` that I'll get rid of later.

Reply:
So I've finally done 2 different commits.
One of them adds a new `focus-visible-013.html` test, which is the same than before but with instructions to run it manually.
And then another commit fixes `focus-visible-007.html`, keeping most of its current design, but removing the wrong parts and the extra checks that are now tested in `focus-visible-013.html`.

Reply:
Thank you so much for this extra work! This cleanup looks excellent.